### PR TITLE
[16.0][MIG] mass_mailing_event_registration_exclude: Migration to version 16.0

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
- Standard migration to v16
- Supersedes https://github.com/OCA/social/pull/1189 due to changes requested not attended.

@Tecnativa

TT45231

@pedrobaeza @chienandalu

### CODE REVIEW

**Comment by carolinafernandez-tecnativa** [comment]
WIP

**Comment by pedrobaeza** [comment]
Can you review the CI?

**Review by fcvalgar — APPROVED** [review]
Functional review

LGTM, thank you @carolinafernandez-tecnativa 

image

image

image

**Review by chienandalu — CHANGES_REQUESTED** [review]
When you're superseding, respect original PR commit attribution and work on top of it. Several changes are missing from that one btw.

**Comment by pedrobaeza** [comment]
Well, sometimes the original PR doesn't have meaningful changes that should be respected, although this PR is now red, so maybe this is one of the cases where you should cherry-pick the commit from the other PR.

**Comment by carolinafernandez-tecnativa** [comment]
> When you're superseding, respect original PR commit attribution and work on top of it. Several changes are missing from that one btw.

@chienandalu Done, I do not understand why tests are failing.
I run the test locally and all test passed:

image

If you see there is an error on that test, but when i run locally i got the same results:

image

File: mass_mailing_event_registration_exclude/models/res_partner.py [hunk 1]
```diff
@@ -10,8 +10,8 @@ class ResPartner(models.Model):
     _inherit = "res.partner"
 
     @api.model
-    def search_count(self, domain):
-        res = super().search_count(domain)
+    def search_count(self, domain, limit=None):
+        res = super().search_count(domain, limit)
```
Comment (chienandalu) [inline mass_mailing_event_registration_exclude/models/res_partner.py]:
```suggestion
        res = super().search_count(domain, limit=limit)
```

File: mass_mailing_event_registration_exclude/models/mailing_contact.py [hunk 2]
```diff
@@ -11,8 +11,8 @@ class MassMailingContact(models.Model):
     _inherit = "mailing.contact"
 
     @api.model
-    def search_count(self, domain):
-        res = super().search_count(domain)
+    def search_count(self, domain, limit=None):
+        res = super().search_count(domain, limit)
```
Comment (chienandalu) [inline mass_mailing_event_registration_exclude/models/mailing_contact.py]:
```suggestion
        res = super().search_count(domain, limit=limit)
```

File: mass_mailing_event_registration_exclude/models/event_registration.py [hunk 3]
```diff
@@ -10,8 +10,8 @@ class EventRegistration(models.Model):
     _inherit = "event.registration"
 
     @api.model
-    def search_count(self, domain):
-        res = super().search_count(domain)
+    def search_count(self, domain, limit=None):
+        res = super().search_count(domain, limit)
```
Comment (chienandalu) [inline mass_mailing_event_registration_exclude/models/event_registration.py]:
```suggestion
        res = super().search_count(domain, limit=limit)
```

File: mass_mailing_event_registration_exclude/models/mailing.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# Copyright 2016 Tecnativa - Antonio Espinosa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import copy
+
+from odoo import fields, models
+
+
+def event_filtered_ids(model, mailing_mailing_id, domain, field="email"):
```
Comment (chienandalu) [inline mass_mailing_event_registration_exclude/models/mailing.py]: It doesn't make much sense to keep this as a global method. Let's put the logic inside the mailing model

**Comment by chienandalu** [comment]
> @chienandalu Done, I do not understand why tests are failing. I run the test locally and all test passed:

You can try to reproduce it:

- Updating your local core code for Odoo/OCB to the latest commit
- Installing other modules in the repo that might be interacting with that condition

**Comment by pedrobaeza** [comment]
Please check CI

**Review by chienandalu — APPROVED** [review]
LGTM Thanks :)

**Pull request merged**